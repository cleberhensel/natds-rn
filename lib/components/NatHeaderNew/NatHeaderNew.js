import React from 'react';
import {
  View,
} from 'react-native';
import PropTypes from 'prop-types';

import { load as loadSvg } from '../../assets/icons/svg';

import {
  HeaderWrapper,
  HeaderTitle,
  IconPress,
  InputIcon,
} from './NatHeaderNew.styles';

export const NatHeaderNew = ({
  title,
  node,
  iconLeft,
  iconRight,
  onPressRight,
  onPressLeft,
}) => {
  if (!title) {
    throw new Error('Title should not empty string');
  }

  const renderIcon = (iconData, iconPress, position) => {
    let styleCustom = {};
    if (position === 'right') {
      styleCustom = { position: 'absolute', right: 0 };
    }
    if (iconData && iconData.svg) {
      const IconComponent = loadSvg(iconData.svg.name);
      const iconSize = (iconData.svg.size) ? iconData.svg.size : 24;
      return (
        <View style={styleCustom}>
          <IconPress onPress={iconPress}>
            <IconComponent width={iconSize} height={iconSize} />
          </IconPress>
        </View>
      );
    }
    if (iconData && iconData.content) {
      return (
        <View style={styleCustom}>
          <IconPress onPress={iconPress}>
            <InputIcon source={iconData.content} />
          </IconPress>
        </View>
      );
    }
    if (iconPress) {
      return (
        <View style={styleCustom}>
          <IconPress onPress={iconPress}>
            <InputIcon source={iconData} />
          </IconPress>
        </View>
      );
    }
    return (
      <View style={styleCustom}>
        <IconPress onPress={iconPress} />
      </View>
    );
  };

  return (
    <HeaderWrapper
      accessible
      accessibilityRole="header"
    >
      { iconLeft !== false && renderIcon(iconLeft, onPressLeft, 'left') }
      <View style={{ justifyContent: 'center', alignItems: 'center' }}>
        { node && node}
        { !node && (
          <HeaderTitle
            accessibilityLabel={`${title}`}
          >
            {title}
          </HeaderTitle>
        )}
      </View>
      { iconRight !== false && renderIcon(iconRight, onPressRight, 'right') }
    </HeaderWrapper>
  );
};

NatHeaderNew.propTypes = {
  title: PropTypes.string,
  node: PropTypes.node,
  iconLeft: PropTypes.string,
  iconRight: PropTypes.string,
  onPress: PropTypes.func,
  onPressRight: PropTypes.func,
  onPressLeft: PropTypes.func,
};

NatHeaderNew.defaultProps = {
  title: null,
  node: null,
  iconLeft: null,
  iconRight: null,
  onPress: () => {},
  onPressRight: () => {},
  onPressLeft: () => {},
};
