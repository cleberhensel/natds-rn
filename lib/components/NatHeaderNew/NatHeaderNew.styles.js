import styled from 'styled-components/native';
import { ThemeNaturaLight } from '../..';

export const HeaderWrapper = styled.View`
  flex: 1;
  width: 100%;
  height: 56;
  padding: 0px;
  position: relative;
  background-color: ${ThemeNaturaLight.palette.background.paper};
  flex-direction: row;
  align-items: stretch;
`;

export const HeaderTitle = styled.Text`
  margin-left: 0px;
  font-size: 20;
  padding: 16px;
  color: ${ThemeNaturaLight.palette.text.primary};
`;

export const IconPress = styled.TouchableOpacity`
  align-self: center;
  flex: 1;
  height: 56;
  width: 56;
  padding: 16px;
`;

export const TouchableOpacity = styled.TouchableOpacity`
  flex: 1;
`;

export const InputIcon = styled.Image`
  height: 24;
  width: 24;
`;
