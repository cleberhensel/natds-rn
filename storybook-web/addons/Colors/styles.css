.colors {
  font-weight: 400;
  line-height: 1.5;
}

.colors__container {
  max-width: 1400px;
  display: flex;
  flex-flow: row wrap;
  padding: 60px;
}

.colors__group {
  display: flex;
  flex-flow: row wrap;
  flex: 0 0 100%;
  margin-bottom: 30px;
}

.colors__group__title {
  font-size: 1.5rem;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-weight: 400;
  color: #555;
  flex: 0 0 100%;
  padding: 0 0 20px;
  text-transform: capitalize;
  margin: 0;
}

.colors__group__list {
  flex: 0 0 100%;
  display: flex;
  flex-flow: row wrap;
  padding: 0;
  margin: 0;
}

.colors__item {
  width: 130px;
  display: flex;
  flex: 0 0 auto;
  position: relative;
  flex-flow: column;
  overflow: hidden;
  padding: 0;
  margin: 0 15px 15px 0;
}

.colors__item__color {
  width: 130px;
  height: 130px;
  margin-bottom: 5px;
}

.colors__item__color-name,
.colors__item__name {
  font-family: Roboto, Arial, Helvetica, sans-serif;
  font-size: 0.875rem;
  line-height: 1rem;
  padding: 0;
  margin: 0;
}

.colors__item__name {
  color: #888;
  margin-bottom: 5px;
}

.colors__item__color-name {
  color: #555;
}
