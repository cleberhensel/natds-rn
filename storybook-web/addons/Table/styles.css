.table__container {
  font-weight: 400;
  line-height: 1.5;
  max-width: 800px;
  width: 100%;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 60px;
}

.table__title {
  font-size: 18px;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 0 20px;
  margin: 0;
}

.table {
  width: 100%;
  border-collapse: collapse;
}

.table__body__row {
  background: #f9f9f9;
  border-bottom: #fff 1px solid;
  padding: 10px;
}

.table__head__item:first-child {
  width: 25%;
}

.table__head__item,
.table__body__item {
  font-size: 0.875rem;
  text-align: left;
}

.table__head__item {
  padding-bottom: 10px;
  text-transform: capitalize;
}

.table__body__item {
  line-height: 1.25rem;
  font-weight: 400;
  height: 20px;
  padding: 10px;
}

.table__body__item__color {
  border: #ddd 2px solid;
  border-radius: 50%;
  width: 16px;
  height: 16px;
  display: inline-block;
  vertical-align: top;
  margin: 2px 10px 0;
}

.table__body__item__color-name {
  line-height: 1rem;
  display: inline-block;
}
