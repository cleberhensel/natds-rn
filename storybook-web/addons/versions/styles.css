.versions {
  font-weight: 400;
  line-height: 1.5;
  font-family: Roboto, Arial, Helvetica, sans-serif;
  color: #555;
  margin: 30px;
}

.versions__type {
  padding: 0 0 10px;
  margin: 0 10px 10px;
}

.versions__type__name {
  border-bottom: rgba(0,0,0,.1) 1px solid;
  font-size: 18px;
  text-transform: capitalize;
  padding: 0 0 10px;
  margin: 0 0 10px;
}

.versions__type__warning {
  font-size: 18px;
  text-transform: capitalize;
  margin-bottom: 10px;
}

.versions__type__button {
  background: rgba(0,0,0,.3);
  border: 0;
  border-radius: 4px;
  font-size: 1rem;
  color: #fff;
  text-transform: capitalize;
  cursor: pointer;
  padding: 7px 14px;
  margin: 0 10px 10px 0;
}

.versions__type__button:hover {
  background: rgba(0,0,0,.2);
}

.versions__type__button:active {
  background: rgba(0,0,0,.25);
}

.versions__type__button--selected,
.versions__type__button--selected:hover {
  background: rgba(0,0,0,.6);
  cursor: default;
}
