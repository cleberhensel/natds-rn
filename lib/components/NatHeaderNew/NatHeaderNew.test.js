import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { NatHeaderNew } from './NatHeaderNew';
import { HeaderTitle } from './NatHeaderNew.styles';

describe('Header component', () => {
  it('Should render container component correctly', () => {
    const header = renderer.create(
      <NatHeaderNew
        title="Title test"
        onPress={jest.fn()}
      />,
    ).toJSON();

    expect(header).toMatchSnapshot();
  });

  it('Should render header component with title', () => {
    const wrapper = shallow(
      <NatHeaderNew
        title="Title test"
        onPress={jest.fn()}
      />,
    );
    expect(wrapper.find(HeaderTitle).text()).toBe('Title test');
  });

  it('Should throw exception when header component title is empty', () => {
    expect(() => shallow(
      <NatHeaderNew
        title=""
        onPress={jest.fn()}
      />,
    )).toThrow('Title should not empty string');
  });
});
