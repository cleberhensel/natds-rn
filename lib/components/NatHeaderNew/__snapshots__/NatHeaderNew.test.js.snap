// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should render container component correctly 1`] = `
<View
  accessibilityRole="header"
  accessible={true}
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
        "flexShrink": 1,
        "height": 56,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
        "position": "relative",
        "width": "100%",
      },
    ]
  }
>
  <View
    style={Object {}}
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 56,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 56,
        }
      }
    >
      <Image
        source={null}
        style={
          Array [
            Object {
              "height": 24,
              "width": 24,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessibilityLabel="Title test"
      style={
        Array [
          Object {
            "color": "#333333",
            "fontSize": 20,
            "marginLeft": 0,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
          },
        ]
      }
    >
      Title test
    </Text>
  </View>
  <View
    style={
      Object {
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "center",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "height": 56,
          "opacity": 1,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 56,
        }
      }
    >
      <Image
        source={null}
        style={
          Array [
            Object {
              "height": 24,
              "width": 24,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
